ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogContainer.tsx

```diff
@@ -77,23 +79,47 @@ export function BrowseFilesDialogContainer(props: BrowseFilesDialogContainerProp
   const [sortKeys, setSortKeys] = useState([]);
   const { username } = useActiveUser();
   const [viewMode, setViewMode] = useState<MediaCardViewModes>(getStoredBrowseDialogViewMode(username));
+  const [fetchingPreselectedItems, setFetchingPreselectedItems] = useState(false);
+  const disableSubmission = fetchingPreselectedItems || (!selectedArray.length && !selectedCard);
 
-  const fetchItems = useCallback(
-    () =>
-      // Since lookahead regex is not supported by opensearch, we are excluding the current path from the search using a
-      // negative filter in a query. This scenario only happens with pages, hence the `withIndex` function wrapping the
-      // current path.
-      search(site, {
-        ...searchParameters,
-        path: `${currentPath}/[^/]+(/index\\.xml)?`,
-        query: `-localId:"${withIndex(currentPath)}"`
-      }).subscribe((response) => {
-        setTotal(response.total);
-        setItems(response.items);
-        setSortKeys(response.facets.map((facet) => facet.name));
-      }),
-    [searchParameters, currentPath, site]
-  );
+  const fetchItems = useCallback(() => {
+    // Since lookahead regex is not supported by opensearch, we are excluding the current path from the search using a
+    // negative filter in a query. This scenario only happens with pages, hence the `withIndex` function wrapping the
+    // current path.
+    search(site, {
+      ...searchParameters,
+      path: `${currentPath}/[^/]+(/index\\.xml)?`,
+      query: `-localId:"${withIndex(currentPath)}"`
+    }).subscribe((response) => {
+      setTotal(response.total);
+      setItems(response.items);
+      setSortKeys(response.facets.map((facet) => facet.name));
+    });
+  }, [searchParameters, currentPath, site]);
+
+  useEffect(() => {
+    const query = preselectedPaths?.map((path) => `localId:"${path}"`).join(' ');
+    setFetchingPreselectedItems(true);
+    search(site, { query }).subscribe({
+      next: ({ items }) => {
+        if (multiSelect) {
+          items.forEach((item) => {
+            setSelectedLookup({ [item.path]: item });
+          });
+        } else {
+          if (items.length) {
+            setSelectedCard(items[0]);
+          }
+        }
+        setFetchingPreselectedItems(false);
+      },
+      error: () => {
+        setFetchingPreselectedItems(false);
+      }
+    });
+    // Interested in renewing the effect when preselectedPaths change, not when the paths array instance changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [site, preselectedPaths.join(''), multiSelect, setSelectedLookup]);
```

Comment:
Let's remove the eslint-disable and the .join and extract the default out of the component to avoid it triggering the effect